proposals/22-03-value-in-the-structured-data.md

```diff
@@ -0,0 +1,511 @@
+will use the feature differently, consider writing separate stories for each.
+Include as much detail as possible so that people can understand the "how" of the system.
+The goal here is to make this feel real for users without getting bogged down.
+-->
+
+#### Story 1
+
+Scenario summary: Replacement the value inside for structured data(json) in the configMap.
+<!--
+A walkthrough of what it will look like for a user to take advantage of the new feature.
+Include the the steps the user will take and samples of the commands they'll run
+and config they'll use.
+-->
+
+kustomize patching overlay is very strong to manage common yaml when using many cluster.\
+But, if you want to set cluster specific change value in the json with configMap data field, you have to replacement whole json file.
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: target-configmap-dev
+data:
+  config.json: |-
+    {"config": {
+      "id": "42",
+      "hostname": "dev.example.com
+    }}
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: target-configmap-prod
+data:
+  config.json: |-
+    {"config": {
+      "id": "42",
+      "hostname": "prod.example.com"
+    }}
+```
+
+So if we can replacement this value in the substring formatted with json, we can easy to overlay this difference.
+
+```yaml
+## source
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: source-configmap
+data:
+  HOSTNAME: www.example.com
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: target-configmap
+data:
+  config.json: |-
+    {"config": {
+      "id": "42",
+      "hostname": "REPLACE_TARGET_HOSTNAME"
+    }}
+```
+
+```yaml
+## replacement
+replacements:
+- source:
+    kind: ConfigMap
+    name: source-configmap
+    fieldPath: data.HOSTNAME
+  targets:
+  - select:
+      kind: ConfigMap
+      name: target-configmap
+    fieldPaths:
+    - data.config\.json
+    options:
+      format: 'json'
+      formatPath: '/config/hostname'
+```
+
+```yaml
+## expected
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: source-configmap
+data:
+  HOSTNAME: www.example.com
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: target-configmap
+data:
+  config.json: '{"config":{"hostname":"www.example.com","id":"42"}}'
+```
+
+#### Story 2
+
+Scenario summary: Merge the two values formatted by structured data(json) with configMapGenerator.
+
+<!--
+A walkthrough of what it will look like for a user to take advantage of the new feature.
+Include the the steps the user will take and samples of the commands they'll run
+and config they'll use.
+-->
+
+Many application needs to be set with json format file. And, That file is handled with configMap when the application is running on kubernetes.\
+So, If kustomize configMapGenerator can overlay to one line inside a configMap data value, A json format file will be simple and easy to handle.
+
+#### Source
+```yaml
+# base/kustomization.yaml
+configMapGenerator:
+- name: demo
+  behavior: merge
```

Comment:
nit: don't set the merge option in the base

Reply:
Sorry, I move it in the overlay.